review/get_review_ids.py

```diff
@@ -0,0 +1,212 @@
+
+    Returns a url (str) with the vertical id to render for review.
+    '''
+    user = crum.get_current_user()
+
+    enroll_user(user, current_course)
+
+    store = modulestore()
+    course_usage_key = store.make_course_usage_key(current_course)
+    course_blocks = get_course_blocks(user, course_usage_key)
+
+    vertical_data = set()
+
+    # Each record corresponds to a problem the user has loaded
+    # in the original course
+    for record in StudentModule.objects.filter(**{'student_id': user.id, 'course_id': current_course, 'module_type': 'problem'}):
+        # record.module_state_key takes the form:
+        # block-v1:{course_id}+type@problem+block@{problem_id}
+        problem_id = str(record.module_state_key).split("@")[-1]
+
+        # Actual logic regarding the record should go here
+        state = json.loads(record.state)
+
+        # The key 'selected' shows up if a problem comes from a
+        # library content module. These cause issues so we skip this.
+        # Issue: Library content contains problems but the CSM brings up
+        # the library content and not the problems within
+        if 'selected' in state:
+            continue
+
+        # parent takes the form: block-v1:{course_id}+type@vertical+block@{vertical_id}
+        parent = course_blocks.get_parents(record.module_state_key)[0]
+        vertical_id = str(parent).split("@")[-1]
+
+        vertical_data.add(vertical_id)
+    if len(vertical_data) < 1:
+        return []
+
+    vertical_to_show = random.sample(vertical_data, 1)[0]
+    review_course_id = REVIEW_COURSE_MAPPING[str(current_course)]
+    return (TEMPLATE_URL.format(course_id=review_course_id,
+                    type='vertical', xblock_id=vertical_to_show))
+
+def enroll_user(user, current_course):
+    '''
+    If the user is not enrolled in the review version of the course,
+    they are unable to see any of the problems. This ensures they
+    are enrolled so they can see review problems.
+
+    Parameters:
+        user (User): the current user interacting with the review xBlock
+        current_course (CourseLocator): The course the learner is currently in
+    '''
+    enrollment_course_id = ENROLLMENT_COURSE_MAPPING[str(current_course)]
+    enrollment_status = get_enrollment(user.username, enrollment_course_id)
+    if not enrollment_status:
+        add_enrollment(user.username, enrollment_course_id)
+    elif not enrollment_status['is_active']:
+        update_enrollment(user.username, enrollment_course_id, is_active=True)
+
+def delete_state_of_review_problem(user, current_course, problem_id):
+    '''
+    Deletes the state of a review problem so it can be used infinitely
+    many times.
+
+    Parameters:
+        user (User): the current user interacting with the review xBlock
+        current_course (CourseLocator): The course the learner is currently in
+        problem_id (str?): The problem id whose state should be cleared
+    '''
+    pass
```

Comment:
And - this method would delete your state/answer in the latter course?

Reply:
Yes. It is something that has been mentioned to me by course teams so that learners can continue to review as much as they would like. I did not consider it important enough to halt v1 so it is left unimplemented for now.